tests/cmdline/params/types/test_code.py

```diff
@@ -7,109 +7,127 @@
-        """
-        identifier = '{}'.format(self.entity_02.label)
-        result = self.param_base.convert(identifier, None, None)
-        self.assertEqual(result.uuid, self.entity_01.uuid)
-
-        identifier = '{}{}'.format(self.entity_02.label, OrmEntityLoader.label_ambiguity_breaker)
-        result = self.param_base.convert(identifier, None, None)
-        self.assertEqual(result.uuid, self.entity_02.uuid)
-
-    def test_ambiguous_label_uuid(self):
-        """
-        Situation: LABEL of entity_03 is exactly equal to UUID of entity_01
-
-        Verify that using an ambiguous identifier gives precedence to the UUID interpretation
-        Appending the special ambiguity breaker character will force the identifier to be treated as a LABEL
-        """
-        identifier = '{}'.format(self.entity_03.label)
-        result = self.param_base.convert(identifier, None, None)
-        self.assertEqual(result.uuid, self.entity_01.uuid)
-
-        identifier = '{}{}'.format(self.entity_03.label, OrmEntityLoader.label_ambiguity_breaker)
-        result = self.param_base.convert(identifier, None, None)
-        self.assertEqual(result.uuid, self.entity_03.uuid)
-
-    def test_entry_point_validation(self):
-        """Verify that when an `entry_point` is defined in the constructor, it is respected in the validation."""
-        identifier = '{}'.format(self.entity_02.pk)
-        result = self.param_entry_point.convert(identifier, None, None)
-        self.assertEqual(result.uuid, self.entity_02.uuid)
-
-        with self.assertRaises(click.BadParameter):
-            identifier = '{}'.format(self.entity_03.pk)
-            result = self.param_entry_point.convert(identifier, None, None)
+@pytest.fixture
+def setup_codes(clear_database_before_test, aiida_localhost):
+    """Create some `Code` instances to test the `CodeParamType` parameter type for the command line infrastructure.
+
+    We create an initial code with a random name and then on purpose create two code with a name that matches exactly
+    the ID and UUID, respectively, of the first one. This allows us to test the rules implemented to solve ambiguities
+    that arise when determing the identifier type.
+    """
+    entity_01 = Code(remote_computer_exec=(aiida_localhost, '/bin/true')).store()
+    entity_02 = Code(remote_computer_exec=(aiida_localhost, '/bin/true'), input_plugin_name='arithmetic.add').store()
+    entity_03 = Code(remote_computer_exec=(aiida_localhost, '/bin/true'), input_plugin_name='templatereplacer').store()
+
+    entity_01.label = 'computer_01'
+    entity_02.label = str(entity_01.pk)
+    entity_03.label = str(entity_01.uuid)
+
+    return entity_01, entity_02, entity_03
+
+
+def test_get_by_id(setup_codes):
+    """Verify that using the ID will retrieve the correct entity."""
+    entity_01, entity_02, entity_03 = setup_codes
+    param = CodeParamType()
```

Comment:
Why not just instantiate this before all the test functions - or make it a fixture?

Reply:
> Why not just instantiate this before all the test functions - or make it a fixture?

What would be the difference? If I have to instantiate it I still need to pass it into each test somehow, which would be best accomplished through a fixture :smile: 
But I agree, I will inject it through a fixture

Reply:
This is done